game/outfits.rpy

```diff
@@ -1,289 +1,2058 @@
-default persistent.jn_natsuki_current_accessory = "hairbands/red"
-default persistent.jn_natsuki_current_eyewear = None
-default persistent.jn_natsuki_current_headgear = None
-default persistent.jn_natsuki_current_necklace = None
-default persistent.jn_natsuki_saved_outfits = {}
-default persistent.jn_natsuki_auto_outfit_change_enabled = True
-
-init python in jn_outfits:
+
+init -2:
+    default persistent.jn_natsuki_auto_outfit_change_enabled = True
+    default persistent.jn_custom_outfits_unlocked = False
+    default persistent.jn_natsuki_outfit_on_quit = "jn_school_uniform"
+    default persistent.jn_outfit_list = {}
+    default persistent.jn_wearable_list = {}
+
+init -1 python in jn_outfits:
+    from Enum import Enum
+    import json
+    import os
     import random
+    import re
     import store
     import store.jn_affinity as jn_affinity
     import store.jn_utils as jn_utils
+    import time
+
+    # Critical file paths
+    __CUSTOM_WEARABLES_DIRECTORY = os.path.join(renpy.config.basedir, "custom_wearables/").replace("\\", "/")
+    __CUSTOM_OUTFITS_DIRECTORY = os.path.join(renpy.config.basedir, "custom_outfits/").replace("\\", "/")
+    __WEARABLE_BASE_PATH = os.path.join(renpy.config.basedir, "game/mod_assets/natsuki/")
+
+    # Ignore the syntax highlighting - "it just works"
+    __RESTRICTED_CHARACTERS_REGEX = "((\.)|(\[)|(\])|(\})|(\{)|(,)|(\!))"
 
-    from store import Natsuki
+    # Lists of all registered outfits/wearables
+    __ALL_WEARABLES = {}
+    __ALL_OUTFITS = {}
 
-    current_outfit_name = None
-    _use_alt_outfit = random.choice(range(1, 3)) == 1
+    _PREVIEW_OUTFIT = None
+    _LAST_OUTFIT = None
 
-    class JNOutfitPreset():
+    _SESSION_NEW_UNLOCKS = list()
+
+    _changes_made = False
+
+    # Wearables being registered via JSON must be one of the following types
+    WEARABLE_CATEGORIES = [
+        "hairstyle",
+        "eyewear",
+        "accessory",
+        "clothes",
+        "headgear",
+        "necklace"
+    ]
+
+    class JNWearable():
         """
-        Describes a complete outfit for Natsuki to wear; including clothing, hairstyle, etc.
-        At minimum, an outfit must consist of clothes and a hairstyle
+        Describes a standalone object that Natsuki can wear.
         """
         def __init__(
             self,
+            reference_name,
             display_name,
+            unlocked,
+            is_jn_wearable
+        ):
+            """
+            Constructor.
+
+            IN:
+                - reference_name - The name used to uniquely identify this wearable and refer to it internally
+                - display_name - The name displayed to the user
+                - unlocked - Whether or not this wearable is selectable to the player on menus
+                - is_jn_wearable - Whether or not this wearable is an official JN wearable. Official wearables cannot be deleted/modified.
+            """
+            self.reference_name = reference_name
```

Comment:
It might be just me, but when I first saw `reference_name` in the code I got confused why it's that long and not just `name`. Now I see it's because you have `display_name`, but I'd maybe change it to `id` or something else.